orca/requester.py

```diff
@@ -1,48 +1,159 @@
 """Module responsible for handling requests to data servers"""
 
 import logging
+import requests
 from xarray import open_mfdataset, open_dataset
 import re
-from math import ceil
 import os
 from datetime import datetime
 
 
 logger = logging.getLogger("scripts")
 
 
-def file_from_opendap(url, outdir, outfile):
+def file_from_opendap(url, threshold, outdir, outfile):
     """Write to file from OPeNDAP link"""
-    urls = bisect_request(url)
+    input_dataset = open_dataset(url)
+    nbytes = input_dataset.nbytes / 2
+    data_vars = input_dataset.data_vars
+    data_vars_with_time = (
+        []
+    )  # Only use data variables with a time component for pivoting in bisect request.
+    time_indices = (
+        []
+    )  # Position of time component for each data variable. Used as pivot for bisecting request.
+    for data_var in data_vars:
+        if "time" in input_dataset[data_var].sizes:
+            data_vars_with_time.append(data_var)
+            time_indices.append(list(input_dataset[data_var].sizes).index("time"))
+    urls = bisect_request(url, threshold, nbytes, data_vars_with_time, time_indices)
 
     logger.debug(f"Downloading from {len(urls)} URL(s): {urls}")
-    dataset = open_mfdataset(urls, combine="nested", concat_dim="time")
+    if len(urls) == 1:
+        dataset = open_dataset(urls[0])
+    else:
+        dataset = open_mfdataset(
+            urls, combine="nested", concat_dim="time", chunks=10 * len(urls)
+        )
     outpath = to_file(dataset, outdir, outfile)
-
     return outpath
 
 
-def to_file(dataset, outdir, outfile=""):
-    """Write netcdf to generated filename"""
+def to_file(dataset, outdir, outfile="", nc=True):
+    """Write data to generated filename"""
     if not outfile:
         now = datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
         outfile = f"orca-output-{now}"
 
     outpath = outdir + outfile
 
     logger.debug("Begin file write")
-    dataset.to_netcdf(outpath)
+    if nc:
+        dataset.to_netcdf(outpath)
+    else:
+        f = open(outpath, "wb")
+        f.write(requests.get(dataset).content)
+        f.close()
     logger.debug("File write complete")
 
-    return outpath
```

Comment:
Why not return `outpath`?

Reply:
It's because the `NamedTemporaryFile` outpath object gets passed to this function, so the file will be updated without having to return it.